src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs

```diff
@@ -29,9 +29,9 @@ internal class GeneralScriptBindingProvider : ScriptBindingProvider
         // The constructor is fixed and ScriptBindingProvider are instantated for us by the Script runtime.
         // Extensions may get registered after this class is instantiated.
         // So we need a final call that lets us get the tooling snapshot of the graph after all extensions are set.
-        public void CompleteInitialization()
+        public void CompleteInitialization(IJobHostMetadataProvider metadataProvider)
         {
-            this._metadataProvider = this.Config.CreateMetadataProvider();
```

Comment:
Yes, although it could have been factored out to a separate change. We had to do some juggling to allow Script to get access to the new IDistributedLockMAnager,   (see [https://github.com/Azure/azure-webjobs-sdk-script/pull/1615](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615) ) 

---
In reply to: [123669490](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123669490) [](ancestors = 123669490)